test/awsE2ESolutionSetup.sh

```diff
@@ -0,0 +1,152 @@
+  echo "  --enable-capture-proxy                           If included, will attempt to enable the capture proxy on all source nodes"
+  echo ""
+  exit 1
+}
+
+stage='aws-integ'
+CREATE_SLR=false
+BOOTSTRAP_REGION=false
+ENABLE_CAPTURE_PROXY=false
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --create-service-linked-roles)
+      CREATE_SLR=true
+      shift # past argument
+      ;;
+    --bootstrap-region)
+      BOOTSTRAP_REGION=true
+      shift # past argument
+      ;;
+    --enable-capture-proxy)
+      ENABLE_CAPTURE_PROXY=true
+      shift # past argument
+      ;;
+    -h|--help)
+      usage
+      ;;
+    -*)
+      echo "Unknown option $1"
+      usage
+      ;;
+    *)
+      shift # past argument
+      ;;
+  esac
+done
+
+if [ "$CREATE_SLR" = true ] ; then
+  create_service_linked_roles
+fi
+
+# Store CDK context for migration solution deployment in variable
+read -r -d '' cdk_context << EOM
+{
+    "stage": "$stage",
+    "vpcId": "<VPC_ID>",
+    "engineVersion": "OS_2.9",
+    "domainName": "opensearch-cluster-aws-integ",
+    "dataNodeCount": 2,
+    "availabilityZoneCount": 2,
+    "mskBrokerNodeCount": 2,
+    "openAccessPolicyEnabled": true,
+    "domainRemovalPolicy": "DESTROY",
+    "migrationAnalyticsServiceEnabled": false,
+    "fetchMigrationEnabled": true,
+    "sourceClusterEndpoint": "<SOURCE_CLUSTER_ENDPOINT>",
+    "dpPipelineTemplatePath": "../../../test/dp_pipeline_aws_integ.yaml"
+}
+EOM
+
+git clone https://github.com/lewijacn/opensearch-cluster-cdk.git || echo "Repo already exists, skipping.."
+cd opensearch-cluster-cdk && git checkout migration-es
+git pull
+npm install
+if [ "$BOOTSTRAP_REGION" = true ] ; then
+  bootstrap_region
+fi
+# Deploy source cluster on EC2 instances
+cdk deploy "*" --require-approval never --c suffix="Migration-Source" --c distVersion="7.10.2" --c distributionUrl="https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-7.10.2-linux-x86_64.tar.gz" --c captureProxyEnabled=true --c captureProxyTarUrl="https://github.com/opensearch-project/opensearch-migrations/releases/download/1.0.0/trafficCaptureProxyServer.x86_64.tar" --c securityDisabled=true --c minDistribution=false --c cpuArch="x64" --c isInternal=true --c singleNodeCluster=true --c networkAvailabilityZones=2 --c dataNodeCount=2 --c managerNodeCount=0 --c serverAccessType="ipv4" --c restrictServerAccessTo="0.0.0.0/0"
+source_endpoint=$(aws cloudformation describe-stacks --stack-name opensearch-infra-stack-Migration-Source --query "Stacks[0].Outputs[?OutputKey==\`loadbalancerurl\`].OutputValue" --output text)
+echo $source_endpoint
+vpc_id=$(aws cloudformation describe-stacks --stack-name opensearch-network-stack --query "Stacks[0].Outputs[?contains(OutputValue, 'vpc')].OutputValue" --output text)
+echo $vpc_id
+
+cdk_context=$(echo "${cdk_context/<VPC_ID>/$vpc_id}")
+cdk_context=$(echo "${cdk_context/<SOURCE_CLUSTER_ENDPOINT>/http://${source_endpoint}}")
+cdk_context=$(echo $cdk_context | jq '@json')
```

Comment:
Is `jq`guaranteed to be installed or should it be listed as a dependency?

Reply:
This script takes on all the dependencies of our solution (and additionally jq). Have added a disclaimer to the top of the file mentioning this